NDTensors/src/lib/SymmetrySectors/src/sector_definitions/su.jl

```diff
@@ -0,0 +1,177 @@
+#
+# Special unitary group SU(N)
+#
+
+using HalfIntegers: HalfInteger, half, twice
+using ...GradedAxes: GradedAxes
+
+struct SU{N,M} <: AbstractSector
+  # l is the first row of the
+  # Gelfand-Tsetlin (GT) pattern describing
+  # an SU(N) irrep
+  # this first row is identical to the Young tableau of the irrep
+  l::NTuple{M,Int}
+
+  # M is there to avoid storing a N-Tuple with an extra zero.
+  # inner constructor enforces M = N - 1
+  # It does NOT check for Young Tableau validity (non-increasing positive integers)
+  function SU{N,M}(t::NTuple{M,Integer}) where {N,M}
+    return N == M + 1 && M > 0 ? new{N,M}(t) : error("Invalid tuple length")
+  end
+end
+
+SU{N}(t::Tuple) where {N} = SU{N,length(t)}(t)
+SU(t::Tuple) = SU{length(t) + 1}(t)  # infer N from tuple length
+
+SymmetryStyle(::Type{<:SU}) = NotAbelianStyle()
+
+sector_label(s::SU) = s.l
+
+groupdim(::SU{N}) where {N} = N
+
+trivial(::Type{<:SU{N}}) where {N} = SU{N}(ntuple(_ -> 0, Val(N - 1)))
+
+fundamental(::Type{<:SU{N}}) where {N} = SU{N}(ntuple(i -> i == 1, Val(N - 1)))
+
+adjoint(::Type{<:SU{N}}) where {N} = SU{N}((ntuple(i -> 1 + (i == 1), Val(N - 1))))
+
+function quantum_dimension(::NotAbelianStyle, s::SU)
+  N = groupdim(s)
+  l = (sector_label(s)..., 0)
+  d = 1
+  for k1 in 1:N, k2 in (k1 + 1):N
+    d *= ((k2 - k1) + (l[k1] - l[k2]))//(k2 - k1)
+  end
+  return Int(d)
+end
+
+function GradedAxes.dual(s::SU)
+  l = sector_label(s)
+  nl = reverse(cumsum((l[begin:(end - 1)] .- l[(begin + 1):end]..., l[end])))
+  return typeof(s)(nl)
+end
+
+function Base.show(io::IO, s::SU)
+  disp = join([string(l) for l in sector_label(s)], ", ")
+  return print(io, "SU(", groupdim(s), ")[", disp, "]")
+end
+
+# display SU(N) irrep as a Young tableau with utf8 box char
+function Base.show(io::IO, ::MIME"text/plain", s::SU)
+  if istrivial(s)   # singlet = no box
+    return print(io, "●")
+  end
+
+  N = groupdim(s)
+  l = sector_label(s)
+  println(io, "┌─" * "┬─"^(l[1] - 1) * "┐")
+  i = 1
+  while i < N - 1 && l[i + 1] != 0
+    println(
+      io,
+      "├─",
+      "┼─"^(l[i + 1] - 1 + (l[i] > l[i + 1])),
+      "┴─"^max(0, (l[i] - l[i + 1] - 1)),
+      "┤"^(l[i] == l[i + 1]),
+      "┘"^(l[i] > l[i + 1]),
+    )
+    i += 1
+  end
+
+  print(io, "└─", "┴─"^max(0, l[i] - 1), "┘")
+  return nothing
+end
+
+#
+# Specializations for the case SU{2}
+#
+
+# optimize implementation
+quantum_dimension(s::SU{2}) = sector_label(s)[1] + 1
+
+GradedAxes.dual(s::SU{2}) = s
+
+function label_fusion_rule(::Type{<:SU{2}}, s1, s2)
+  irreps = collect(SU{2}((i,)) for i in (abs(s1[1] - s2[1])):2:(s1[1] + s2[1]))
```

Comment:
```suggestion
  irreps = [SU{2}((i,)) for i in (abs(s1[1] - s2[1])):2:(s1[1] + s2[1])]
```
I'm pretty sure that's the same thing but more compact, using the comprehension syntax (https://docs.julialang.org/en/v1/manual/arrays/#man-comprehensions).